src/zarr/store/remote.py

```diff
@@ -1,103 +1,190 @@
 from __future__ import annotations
 
+from collections.abc import AsyncGenerator
 from typing import TYPE_CHECKING, Any
 
+import fsspec
+
 from zarr.abc.store import Store
-from zarr.buffer import Buffer, BufferPrototype
+from zarr.buffer import Buffer, BufferPrototype, default_buffer_prototype
 from zarr.common import OpenMode
 from zarr.store.core import _dereference_path
 
 if TYPE_CHECKING:
     from fsspec.asyn import AsyncFileSystem
     from upath import UPath
 
+    from zarr.buffer import Buffer
+    from zarr.common import BytesLike
+
 
 class RemoteStore(Store):
+    # based on FSSpec
     supports_writes: bool = True
     supports_partial_writes: bool = False
     supports_listing: bool = True
 
-    root: UPath
+    _fs: AsyncFileSystem
+    path: str
+    exceptions: tuple[type[Exception], ...]
```

Comment:
I think we should call this attribute `allowed_exceptions` to match the kwarg and to more clearly convey what it does.